src/fosslight_dependency/package_manager/Yarn.py

```
@@ -59,23 +60,43 @@ def start_license_checker(self):
+                if is_pnp_mode:
+                    # Force node-modules linker via env var (Yarn Berry supports YARN_<SETTING> env vars)
+                    logger.info("Attempting to create node_modules for PnP project using YARN_NODE_LINKER=node-modules...")
+                    yarn_install_cmd = 'YARN_NODE_LINKER=node-modules yarn install --mode=skip-build'
+                else:
+                    yarn_install_cmd = 'yarn install --mode=skip-build'
             else:
                 yarn_install_cmd = 'yarn install --production --ignore-scripts'
-                logger.info(f"Executing: {yarn_install_cmd}")
+            logger.info(f"Executing: {yarn_install_cmd}")
 
             result = subprocess.run(yarn_install_cmd, shell=True, capture_output=True, text=True)
```

Ruff (0.15.14)

[error] `subprocess` call with `shell=True` identified, security issue

(S602)